app/src/main/java/com/macbitsgoa/ard/activities/AuthActivity.java

```diff
@@ -0,0 +1,81 @@
+package com.macbitsgoa.ard.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+
+import com.macbitsgoa.ard.R;
+import com.macbitsgoa.ard.activities.helpers.AuthHelper;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
+ * AuthActivity authenticates user.
+ * The backend functions are implemented in {@link AuthHelper} class.
+ * Currently implemented:
+ * 1. Sign in with Google
+ *
+ * @author Rushikesh Jogdand
+ */
+public class AuthActivity extends BaseActivity
+        implements View.OnClickListener {
+
+    /**
+     * Request code for google sign in.
+     */
+    public static final int RC_GOOGLE_SIGN_IN = 17;
+
+    /**
+     * AuthHelper instance to handle backend functions.
+     */
+    protected AuthHelper helper;
+
+    /**
+     * Google Sign In Button.
+     */
+    @BindView(R.id.btn_google_sign_in)
+    Button googleSignInButton;
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_auth);
+        ButterKnife.bind(this);
+        googleSignInButton.setOnClickListener(this);
+        helper = new AuthHelper(this);
+    }
+
+    @Override
+    public void onClick(final View v) {
+        if (v.getId() == R.id.btn_google_sign_in) {
```

Comment:
since there is only 1 button for now, might as well not bother with checking it's ID.
will avoid reducing coverage